src/Authentication/Authentication/custom/ParameterCompleters.ps1

```diff
@@ -0,0 +1,26 @@
+# ------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+# ------------------------------------------------------------------------------
+
+Set-StrictMode -Version 2
+
+. "$psscriptroot/common/Permissions.ps1"
+
+$scopesParameterBlock = {
+    param($commandName,
+    $parameterName,
+    $wordToComplete,
+    $commandAst,
+    $fakeBoundParameters
+    )
+
+    $permissions = @()
+    $permissionsData = Permissions_GetPermissionsData
+    $permissionsOAuthQuery = Permissions_GetOauthData $permissionsData
```

Comment:
nit: we don't need to hold on to `$permissionsData` as we only pass it to `Permissions_GetOauthData`.
```suggestion
    $permissionsOAuthQuery = Permissions_GetOauthData (Permissions_GetPermissionsData)
```